app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -189,11 +261,121 @@ export default Controller.extend(CanCheckEmails, {
       return this.model.silence();
     },
     deleteAllPosts() {
-      return this.model.deleteAllPosts();
+      let deletedPosts = 0;
+      let deletedPercentage = 0;
+      const user = this.model;
+      const message = I18n.messageFormat(
+        "admin.user.delete_all_posts_confirm_MF",
+        {
+          POSTS: user.get("post_count"),
+          TOPICS: user.get("topic_count"),
+        }
+      );
+
+      const performDelete = () => {
+        this.model
+          .deleteAllPosts()
+          .then(({ posts_deleted }) => {
+            if (posts_deleted === 0) {
+              user.set("post_count", 0);
+              bootbox.hideAll();
+            } else {
+              deletedPosts += posts_deleted;
+              deletedPercentage = Math.floor(
+                (deletedPosts * 100) / user.get("post_count")
+              );
+              $(".delete-posts-progress .progress-bar > span").css({
```

Comment:
We shouldn't use jQuery in controllers either. Is there any way we could implement this using regular ember?